tests/template_tests/filter_tests/test_truncatewords_html.py

```diff
@@ -24,15 +24,15 @@ def test_truncate2(self):
             truncatewords_html(
                 '<p>one <a href="#">two - three <br>four</a> five</p>', 4
             ),
-            '<p>one <a href="#">two - three …</a></p>',
+            '<p>one <a href="#">two - three <br> …</a></p>',
         )
 
     def test_truncate3(self):
         self.assertEqual(
             truncatewords_html(
                 '<p>one <a href="#">two - three <br>four</a> five</p>', 5
             ),
-            '<p>one <a href="#">two - three <br>four …</a></p>',
+            '<p>one <a href="#">two - three <br>four</a> …</p>',
```

Comment:
This is different, but I think a change for the better. Not sure whether we need a backward incompatibility note just to warn that there may be small differences, but that, in general, the output is better and that the implementation is more robust?